sqlccl: actually sanitize backup descriptions

benesch opened the pull request:
We weren't properly sanitizing credentials in incremental URLs.

h/t @dt

benesch committed:
sqlccl: actually sanitize backup descriptions

We weren't properly sanitizing credentials in incremental URLs.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/16370)
<!-- Reviewable:end -->

danhhz commented:
:lgtm: Test?

Also, wanna cherry-pick this into 1.0.2 (or 1.0.3)?

benesch committed:
sqlccl: test that INCREMENTAL FROM URLs are sanitized

We weren't properly sanitizing credentials specified as `INCREMENTAL
FROM` URLs in backups. This was fixed in 9df2877 and backported to the
1.0 branch; this commit adds a test to prevent future regressions.

benesch commented:
Updated with a test! PTAL. And yup, I'll backport the first commit to 1.0.x.

danhhz commented:
Reviewed 1 of 1 files at r1, 1 of 1 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks pending.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/16370)*
<!-- Sent from Reviewable.io -->
